### PR TITLE
Enable the `:set-timezone` driver feature; allow to change the reporting timezone in the Localization settings

### LINKED ISSUE #200: Add fully fledged time zones support
Despite ClickHouse itself featuring great time zone(s) support, these two driver features were _never enabled_:

```
:set-timezone
:convert-timezone
```

There was a draft attempt from me recently, with this snippet commented out and a few other parts missing: https://github.com/ClickHouse/metabase-clickhouse-driver/blob/b19928448adc5961bd37a8080b271f57756f67e1/src/metabase/driver/clickhouse_qp.clj#L320-L327

However, a significant amount of Metabase timezone-related tests fail, most of the time because of hardcoded branches for specific drivers.

Examples of such tests: 
- https://github.com/metabase/metabase/blob/master/test/metabase/query_processor_test/date_time_zone_functions_test.clj#L273-L304
- https://github.com/metabase/metabase/blob/master/test/metabase/query_processor_test/date_time_zone_functions_test.clj#L190-L198
- https://github.com/metabase/metabase/blob/master/test/metabase/query_processor_test/date_time_zone_functions_test.clj#L705-L722

Related Metabase issue to remove the hardcoded parts: https://github.com/metabase/metabase/issues/26807

### PR DESCRIPTION
**Pull request by slvrtrn**
## Summary

This PR enables the `:set-timezone` Metabase feature so that it is now possible to change the reporting timezone in Admin -> Localization settings.

image

Resolves #200. 

Excluded `:convert-timezone` from the scope of this PR, as `:set-timezone` seems to be of a higher priority, and proper `:convert-timezone` implementation requires additional investigation, as some tests still indicate that the implementation (currently, commented out) is not entirely correct.

`:convert-timezone` issue for tracking: https://github.com/ClickHouse/metabase-clickhouse-driver/issues/254

## Checklist
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by slvrtrn** [comment]
Excluded `:convert-timezone` from the scope of this PR, as `:set-timezone` seems to be of a higher priority, and proper `:convert-timezone` implementation requires additional time.

Created a new issue: https://github.com/ClickHouse/metabase-clickhouse-driver/issues/254

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# 1.50.1
+
+### New features
+
+* Enabled `:set-timezone` ([docs](https://www.metabase.com/docs/latest/configuring-metabase/localization#report-timezone)) Metabase feature in the driver. ([#200](https://github.com/ClickHouse/metabase-clickhouse-driver/issues/200))
+
+### Other
+
+* The driver now uses [`session_timezone` ClickHouse setting](https://clickhouse.com/docs/en/operations/settings/settings#session_timezone). This is necessary to support the `:set-timezone` feature; however, this setting [was introduced in 23.6](https://clickhouse.com/docs/en/whats-new/changelog/2023#236), which makes it the minimal required ClickHouse version to work with the driver.
```
Comment (mshustov) [inline CHANGELOG.md:9]: >  The setting takes precedence over the globally configured (server-level) implicit time zone. 

I hope every user reads this warning. Should we add it to the changelog?

**Review by mshustov — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (slvrtrn) [reply CHANGELOG.md:9]: I think I will add some additional explanation when I will rework the README and the website docs.

**Comment by frankyso** [comment]
Hi All,
I wanted to extend a huge thank you to you and the team for enabling the timezone function in Metabase! This enhancement is incredibly useful for our workflows.

I also wanted to ask about the timeline for implementing this feature in Metabase Cloud. Could you please provide any insights or updates on when we can expect it to be available there?

**Comment by Tony-metabase** [comment]
That functionality will be available on Metabase version 50 ... Metabase Cloud is still running 1.49 ... The Cloud would be updated in the coming weeks. You should get a notification when that happens

**Pull request merged**